src/Utility/Test/TestCoordinateMapping.C

```diff
@@ -26,162 +26,144 @@
-}
-
-static Matrix3D readMatrices(size_t r, size_t c, size_t k, const std::string& file)
-{
-  Matrix3D result(r,c,k);
-  std::ifstream f(file);
-  for (size_t n=1;n<=k;++n)
-    for (size_t i=1;i<=r;++i)
-      for (size_t j=1;j<=c;++j)
-        f >> result(i,j,n);
-
-  return result;
-}
-
-auto&& CHECK_MATRICES_EQUAL = [](const Matrix& A, const std::string& path)
-{
-  Matrix B = readMatrix(A.rows(), A.cols(), path);
-  for (size_t i=1;i<=A.rows();++i)
-    for (size_t j=1;j<=A.cols();++j)
-      ASSERT_NEAR(A(i,j), B(i,j), 1e-13);
-};
-
-
-auto&& CHECK_MATRICES3D_EQUAL = [](const Matrix3D& A, const std::string& path)
-{
-  Matrix3D B = readMatrices(A.dim(1), A.dim(2), A.dim(3), path);
-  for (size_t i=1;i<=A.dim(1);++i)
-    for (size_t j=1;j<=A.dim(2);++j)
-      for (size_t k=1;k<=A.dim(3);++k)
-        ASSERT_NEAR(A(i,j,k), B(i,j,k), 1e-13);
-};
-
 
 TEST(TestCoordinateMapping, Jacobian1D)
 {
   SIM1D sim(1);
-  sim.createDefaultModel();
+  ASSERT_TRUE(sim.createDefaultModel());
   ASMs1D& p = static_cast<ASMs1D&>(*sim.getPatch(1));
-  p.uniformRefine(3);
-  sim.preprocess();
+  ASSERT_TRUE(p.uniformRefine(3));
+  ASSERT_TRUE(sim.createFEMmodel());
 
   Vector N;
-  utl::matrix<Real> dNdU;
+  Matrix X, dNdU;
   p.extractBasis(0.25, N, dNdU);
-  utl::matrix<Real> X;
   p.getElementCoordinates(X, 2);
 
-  utl::matrix<Real> J;
-  utl::matrix<Real> dNdX;
+  Matrix J, dNdX;
   Real det = utl::Jacobian(J, dNdX, X, dNdU, true);
-  ASSERT_FLOAT_EQ(det, 1.0);
-  ASSERT_FLOAT_EQ(J(1,1), 1.0);
-  CHECK_MATRICES_EQUAL(dNdX, "src/Utility/Test/refdata/Jacobian1D_dNdX.asc");
+
+  EXPECT_FLOAT_EQ(det, 1.0);
+  EXPECT_FLOAT_EQ(J(1,1), 1.0);
+  EXPECT_EQ(dNdX.rows(), 2U);
+  EXPECT_EQ(dNdX.cols(), 1U);
+  EXPECT_FLOAT_EQ(dNdX(1,1),-4.0);
+  EXPECT_FLOAT_EQ(dNdX(2,1), 4.0);
 }
 
 
 TEST(TestCoordinateMapping, Hessian1D)
 {
   SIM1D sim(1);
-  sim.createDefaultModel();
+  ASSERT_TRUE(sim.createDefaultModel());
   ASMs1D& p = static_cast<ASMs1D&>(*sim.getPatch(1));
-  p.raiseOrder(2);
-  p.uniformRefine(3);
-  sim.preprocess();
+  ASSERT_TRUE(p.raiseOrder(2));
+  ASSERT_TRUE(p.uniformRefine(3));
+  ASSERT_TRUE(sim.createFEMmodel());
 
   Vector N;
-  utl::matrix<Real> dNdU;
+  Matrix X, dNdU;
   Matrix3D d2Ndu2;
   p.extractBasis(0.25, N, dNdU, d2Ndu2);
-  utl::matrix<Real> X;
   p.getElementCoordinates(X, 2);
 
-  utl::matrix<Real> J;
-  utl::matrix<Real> dNdX;
-  Matrix3D d2NdX2;
-  Matrix3D H;
+  Matrix J, dNdX;
+  Matrix3D H, d2NdX2;
   Real det = utl::Jacobian(J, dNdX, X, dNdU, true);
   utl::Hessian(H, d2NdX2, J, X, d2Ndu2, dNdX);
 
-  ASSERT_FLOAT_EQ(det, 1.0);
-  ASSERT_FLOAT_EQ(J(1,1), 1.0);
-  CHECK_MATRICES_EQUAL(dNdX, "src/Utility/Test/refdata/Hessian1D_dNdX.asc");
-  CHECK_MATRICES3D_EQUAL(d2NdX2, "src/Utility/Test/refdata/Hessian1D_d2NdX2.asc");
-  ASSERT_NEAR(H(1,1,1), 0.0, 1e-12);
+  EXPECT_FLOAT_EQ(det, 1.0);
+  EXPECT_FLOAT_EQ(J(1,1), 1.0);
+  EXPECT_EQ(dNdX.rows(), 4U);
+  EXPECT_EQ(dNdX.cols(), 1U);
+  EXPECT_FLOAT_EQ(dNdX(1,1),-3.0);
```

Comment:
okay but i still think separation of data and code is better (ie inline a table).

Reply:
OK, can inline tables instead to make more readable (at least for the non-trivial ones).